statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,254 @@
+import statsmodels.tsa.base.tsa_model as tsbase
+import numpy as np
+import scipy
+
+# VECM class: for known or unknown VECM
+
+class VECM(tsbase.TimeSeriesModel):
+    r"""
+    Fit VECM process and do lag order selection
+
+    .. math:: y_t = A_1 y_{t-1} + \ldots + A_p y_{t-p} + u_t
+
+    Parameters
+    ----------
+    endog : array-like
+        2-d endogenous response variable. The independent variable.
+    dates : array-like
+        must match number of rows of endog
+
+    References
+    ----------
+    Lutkepohl (2005) New Introduction to Multiple Time Series Analysis
+    """#TODO: docstring + implementation
+    
+    # TODO: implementation
+    def __init__(self, endog, dates=None, freq=None, missing='none'):
+        super(VECM, self).__init__(endog, None, dates, freq, missing=missing)
+        if self.endog.ndim == 1:
+            raise ValueError("Only gave one variable to VECM")
+        self.y = self.endog #TODO delete this line if y not necessary
+        self.neqs = self.endog.shape[1]
+
+    def predict(self, params, start=None, end=None, lags=1, trend='c'):
+        """
+        Returns in-sample predictions or forecasts
+        """#TODO: docstring + implementation
+        pass
+
+    def fit(self, max_diff_lags=None, method='ml', ic=None, 
+            deterministic_terms='', verbose=False, coint_rank=None):
+        """
+        Fit the VECM
+
+        Parameters
+        ----------
+        max_diff_lags : int
+            Maximum number of lags to check for order selection, defaults to
+            12 * (nobs/100.)**(1./4), see select_order function ##### ##### ##### #####
+        method : {'ls', 'egls', 'ml'}
+            Estimation method to use.
+        ic : {'aic', 'fpe', 'hqic', 'bic', None}
+            Information criterion to use for VECM order selection.
+            aic : Akaike
+            fpe : Final prediction error
+            hqic : Hannan-Quinn
+            bic : Bayesian a.k.a. Schwarz
+        deterministic terms, str {"", "c", "lt", "s"}
+            "" - no deterministic terms
```

Comment:
In ARMA and similar we call it "nc" for no constant.

(based on johansen cointegration tests, we might need 5 or 6 versions, depending on whether the constant, deterministic part is in the cointegrating vector or in the difference equation.)